### PR TITLE
qdrant.ts: add support for QDRANT_API_KEY 

### PR DESCRIPTION
**Pull request by NirantK**
Adds support for `apiKey`

From the official [Qdrant JS Client docs](https://github.com/qdrant/qdrant-js): 

```javascript
import {QdrantClient} from '@qdrant/js-client-rest';

// TO connect to Qdrant running locally
const client = new QdrantClient({url: 'http://127.0.0.1:6333'});

// or connect to Qdrant Cloud
const client = new QdrantClient({
    url: 'https://xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.us-east-0-1.aws.cloud.qdrant.io',
    apiKey: '<your-api-key>',
});
```

### CODE REVIEW

File: langchain/src/vectorstores/qdrant.ts [hunk 1]
```diff
@@ -32,17 +33,24 @@ export class QdrantVectorStore extends VectorStore {
...
-      });
-
+    // apiKey is necessary for Qdrant Cloud, but not self hosted
+    if (!args.client && apiKey) {
+        this.client =
+        args.client ||
+        new QdrantClient({
+            url,
+            apiKey,
+        });
```
Comment (generall) [inline langchain/src/vectorstores/qdrant.ts:49]: Empty api key won't hurt, you can drop the condition and just use the version with api key param

**Comment by dhruv-anand-aintech** [comment]
LGTM. please update langchain documentation (if applicable), and address above comment

File: langchain/src/vectorstores/qdrant.ts (hunk 1 above)
Comment (NirantK) [reply langchain/src/vectorstores/qdrant.ts]: Done!

**Review by nfcampos — APPROVED** [review]
Thanks lgtm

**Pull request merged**